lib/workers/repository/update/branch/get-updated.ts

```diff
@@ -526,3 +548,168 @@ async function applyManagerBumpPackageVersion(
+        };
+      }
+      try {
+        if (res.file.previousContents) {
+          parsedPreviousContents = yaml.load(
+            res.file.previousContents.toString(),
+          );
+        }
+      } catch (e) {
+        parsedPreviousContents = {
+          error: 'Failed to parse previousContents as YAML',
+          details: e,
+        };
+      }
+      // Extract sourcerpm fields from both YAMLs
+      const getSourcerpms = (parsed: any): string[] => {
+        if (!parsed) {
+          return [];
+        }
+        const rpms: string[] = [];
+        const search = (obj: any): void => {
+          if (Array.isArray(obj)) {
+            obj.forEach(search);
+          } else if (obj && typeof obj === 'object') {
+            for (const [key, value] of Object.entries(obj)) {
+              if (key === 'sourcerpm' && typeof value === 'string') {
+                rpms.push(value);
+              } else {
+                search(value);
+              }
+            }
+          }
+        };
+        search(parsed);
+        return rpms;
+      };
+      const newSourcerpms = getSourcerpms(parsedContents);
+      const oldSourcerpms = getSourcerpms(parsedPreviousContents);
+      // Map by name for easy comparison
+      const parseRpm = (
+        s: string,
+      ): { name: string; version: string } | null => {
+        const m = rpmRegex.exec(s);
+        if (!m?.groups) {
+          return null;
+        }
+        return {
+          name: m.groups.name,
+          version: m.groups.version,
+        };
+      };
+      const oldMap = new Map<string, string>();
+      for (const rpm of oldSourcerpms) {
+        const parsed = parseRpm(rpm);
+        if (parsed) {
+          oldMap.set(parsed.name, parsed.version);
+        }
+      }
+      for (const rpm of newSourcerpms) {
+        const parsed = parseRpm(rpm);
+        if (parsed) {
+          const oldVersion = oldMap.get(parsed.name);
+          if (oldVersion && oldVersion !== parsed.version) {
+            rpmVersionChanges.push({
+              name: parsed.name,
+              oldVersion,
+              newVersion: parsed.version,
+            });
+          }
+        }
+      }
+    }
+    return rpmVersionChanges;
+  });
+  return changes.length > 0 ? changes : null;
+}
+
+// this will be used for CVE information, but just for demonstration purposes
+function addRpmtestPrBodyNotes(
+  config: BranchConfig,
+  parsedResults:
+    | { name: string; oldVersion: string; newVersion: string }[]
+    | null,
+): void {
+  if (!parsedResults || parsedResults.length === 0) {
+    return;
+  }
+  // Assume there is always exactly one upgrade
+  const upgrade = Array.isArray((config as any).upgrades)
+    ? (config as any).upgrades[0]
+    : undefined;
+  if (!upgrade) {
+    return;
+  }
+  if (!Array.isArray(upgrade.prBodyNotes)) {
+    upgrade.prBodyNotes = [];
+  }
+  for (const { name, oldVersion, newVersion } of parsedResults) {
+    upgrade.prBodyNotes.push(
```

Comment:
Could we modify the `config: BranchConfig` and push sort of "fake" upgrades in there to fill the upgrade table in the PR? Sounds too hacky, but I'm just wondering.